### PR TITLE
Add access rule resource

### PR DESCRIPTION
**Pull request by SimonBarendse**
Adds an access rule resource to manage access levels in Terraform. Together with service resources that we'll add as well, application access can be fully managed in Terraform.

### CODE REVIEW

File: secrethub/resource_access_rule.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+func resourceAccessRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAccessRuleSet,
+		Read:   resourceAccessRuleRead,
+		Update: resourceAccessRuleSet,
+		Delete: resourceAccessRuleDelete,
+		Schema: map[string]*schema.Schema{
+			"dir_path": {
```
Comment (SimonBarendse) [inline secrethub/resource_access_rule.go]: Maybe just `path` here, instead of `dir_path`?

File: secrethub/resource_access_rule.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The path of the directory on which the permission holds.",
+			},
+			"account_name": {
```
Comment (SimonBarendse) [inline secrethub/resource_access_rule.go]: Not completely sure what name to use here. For now it accepts account names (usernames or service IDs), but I can imagine we'll add setting access rules by account ID later.

File: secrethub/resource_access_rule.go (hunk 1 above)
Comment (florisvdg) [reply secrethub/resource_access_rule.go]: Depends, does it also support secrets?

File: secrethub/resource_access_rule.go (hunk 2 above)
Comment (florisvdg) [reply secrethub/resource_access_rule.go]: Maybe just `account`, and change the description to: `The username or service ID...`

File: secrethub/resource_access_rule.go (hunk 1 above)
Comment (SimonBarendse) [reply secrethub/resource_access_rule.go]: Nope, just directories

Comment (SimonBarendse) [reply secrethub/resource_access_rule.go]: And what about just `dir`? Or does that make it unclear whether we want the path or the ID? We might want to accept the ID as well later, but we currently do not yet do so.

File: secrethub/resource_access_rule.go (hunk 2 above)
Comment (SimonBarendse) [reply secrethub/resource_access_rule.go]: And when we add support for supplying IDs? Will we supply those on `account` as well? I believe we can only determine the difference between UUID and username based on the length. Are we comfortable setting these maxima as hard requirements, so we can keep this length check as a way to differentiate the two? Then we cannot relax these constraints later.

File: secrethub/resource_access_rule.go (hunk 1 above)
Comment (florisvdg) [reply secrethub/resource_access_rule.go]: Let's stick with `dir_path`, I'd say. Not sure a dir ID is something we'd like to expose to the users.

File: secrethub/resource_access_rule.go (hunk 2 above)
Comment (SimonBarendse) [reply secrethub/resource_access_rule.go]: Discussed this in-person. We also use `account_name` in the API for referring to both username and service ID. We'll also do so here.

File: secrethub/resource_access_rule.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	provider := m.(providerMeta)
+	client := *provider.client
+
+	path := d.Get("dir_path").(string)
+	permission := d.Get("permission").(string)
+	account := d.Get("account_name").(string)
+
+	_, err := client.AccessRules().Set(path, permission, account)
+
+	d.SetId(fmt.Sprintf("path: %s, account_name: %s", path, account))
```
Comment (florisvdg) [inline secrethub/resource_access_rule.go]: Something like this may be nicer, more 'id-like': `path + ":" + account`

**Pull request merged**